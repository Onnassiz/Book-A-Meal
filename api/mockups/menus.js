const menus = [
  {
    name: 'Sunday\'s blast',
    timestamp: 1524960000,
    meals: [
      {
        id: 6,
        name: 'Yum - Pork',
        description: 'Slice Thin Beef Mixed With Lettuce, Thread Noodle, Lime, Onion, Carrot, Cilantro. Topped with Peanuts. Choice of Beef, Pork or Seafood.',
        price: 2100,
        category: 'Hot Meal',
        image: '6.jpg',
      },
      {
        id: 3,
        name: 'Vegetable Fried Rice (Khao Phat Phak)',
        description: 'Stir Fried with Eggs, Chinese Broccoli, Tofu and Onion. Served with Cucumber and Lime.',
        price: 1500,
        category: 'Snack',
        image: '3.jpg',
      },
      {
        id: 4,
        name: 'Shrimp Frid Rice',
        description: 'Stir Fried Rice with Shrimp, Egg, Peas, Corn, Bean, Carrots, Onion. Topped with Cucumber and Lime.',
        price: 2000,
        category: 'Snack',
        image: '4.jpg',
      },
    ],
  },
  {
    name: 'Monday\'s delight',
    timestamp: 1525046400,
    meals: [
      {
        id: 6,
        name: 'Yum - Pork',
        description: 'Slice Thin Beef Mixed With Lettuce, Thread Noodle, Lime, Onion, Carrot, Cilantro. Topped with Peanuts. Choice of Beef, Pork or Seafood.',
        price: 2100,
        category: 'Hot Meal',
        image: '6.jpg',
      },
      {
        id: 3,
        name: 'Vegetable Fried Rice (Khao Phat Phak)',
        description: 'Stir Fried with Eggs, Chinese Broccoli, Tofu and Onion. Served with Cucumber and Lime.',
        price: 1500,
        category: 'Snack',
        image: '3.jpg',
      },
      {
        id: 4,
        name: 'Shrimp Frid Rice',
        description: 'Stir Fried Rice with Shrimp, Egg, Peas, Corn, Bean, Carrots, Onion. Topped with Cucumber and Lime.',
        price: 2000,
        category: 'Snack',
        image: '4.jpg',
      },
    ],
  },
  {
    name: 'Tuesday\'s delight',
    timestamp: 1524873600,
    meals: [
      {
        id: 1,
        name: 'Chicken Pa-Naeng - Beef',
        description: 'Red Curry with Peanut Sauce, Bamboo Shoot, Ginger, Bell Pepper, Green Bean, Galanga and Lemon Leaves. Choice of Chicken, Beef, Pork, Tofu or Shrimp.',
        price: 2300,
        category: 'Hot Meal',
        image: '1.jpg',
      },
      {
        id: 2,
        name: 'Squash Curry - Tofu',
        description: 'Red and Yellow Curry Coconut Milk with Squash, Potatoes, Green Bean & Bamboo Shoot. Choice of Chicken, Beef, Pork, Tofu or Shrimp.',
        price: 1200,
        category: 'Lunch',
        image: '2.jpg',
      },
      {
        id: 4,
        name: 'Shrimp Frid Rice',
        description: 'Stir Fried Rice with Shrimp, Egg, Peas, Corn, Bean, Carrots, Onion. Topped with Cucumber and Lime.',
        price: 2000,
        category: 'Snack',
        image: '4.jpg',
      },
    ],
  },
];

module.exports = menus;
