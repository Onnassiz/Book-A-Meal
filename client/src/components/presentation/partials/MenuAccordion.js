import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Pagination from 'react-js-pagination';

import {
  AccordionItem,
  AccordionItemTitle,
  AccordionItemBody,
} from 'react-accessible-accordion';

class MenuAccordion extends Component {
  constructor(props) {
    super(props);
    this.state = {
      menu: {},
      showMeals: false,
      activePage: 1,
      thisPageMeals: [],
    };
    this.handlePageChange = this.handlePageChange.bind(this);
    this.showMeals = this.showMeals.bind(this);
  }

  componentWillMount() {
    const { menu } = this.props;
    this.setState({ menu, thisPageMeals: menu.mealsArray });
  }

  handlePageChange(pageNumber) {
    const { getMealsInMenu } = this.props;
    const offset = (pageNumber - 1) * 5;

    getMealsInMenu(this.state.menu, offset).then((response) => {
      if (response.response.status === 200) {
        this.setState({
          activePage: pageNumber,
          thisPageMeals: response.response.data,
        });
      }
    });
  }

  showMeals() {
    const { getMealsInMenu } = this.props;
    getMealsInMenu(this.state.menu, 0).then((response) => {
      if (response.response.status === 200) {
        this.setState({
          activePage: 1,
          thisPageMeals: response.response.data,
          showMeals: true,
        });
      }
    });
  }

  renderAccordionTitle() {
    return (
      <AccordionItemTitle>
        <h3 className="u-position-relative">
          {new Date(this.state.menu.date).toDateString()}
          <div className="accordion__arrow" role="presentation" />
        </h3>
        <div style={{ fontSize: 13, marginTop: 5 }}>{this.state.menu.mealsCount}{this.state.menu.mealsCount > 1 ? ' meals' : ' meal'}</div>
      </AccordionItemTitle>
    );
  }

  renderPagination() {
    return (
      <div>{this.state.menu.mealsCount > 5 ?
        <div style={{ textAlign: 'center' }}>
          <Pagination
            hideDisabled
            activePage={this.state.activePage}
            itemsCountPerPage={5}
            totalItemsCount={this.state.menu.mealsCount}
            pageRangeDisplayed={5}
            onChange={this.handlePageChange}
          />
          <hr />
        </div> : ''}
      </div>
    );
  }

  renderMeals() {
    return (
      <div>
        { this.state.thisPageMeals.length > 0 ? this.state.thisPageMeals.map(meal => (
          <div className="menu_item" key={meal.id}>
            <h3>{ meal.name }</h3>
            <div>
              <p><b>Description:</b> {meal.description}</p>
              <p><b>Category: </b>{ meal.category }</p>
              <p><b>Price: </b>{ meal.price }</p>
            </div>
          </div>)) : ''}
        { this.renderPagination() }
      </div>
    );
  }

  render() {
    return (
      <AccordionItem>
        {this.renderAccordionTitle()}
        <AccordionItemBody>
          {this.state.showMeals ?
            this.renderMeals() :
            <div className="show-meals">
              <button className="button" onClick={this.showMeals}><i className="ion-ios-eye" /> Show Meals</button>
            </div>}
          {!this.props.showOpsButtons ? '' :
          <div>
            <div style={{ textAlign: 'center' }}>
              <button style={{ width: 110 }} onClick={this.props.toggleShowDeleteModal} className="button-error"><i className="ion-ios-trash" /> Delete</button>
              <button style={{ width: 110 }} onClick={this.props.toggleUpdateModal} className="button"><i className="ion-ios-compose-outline" /> Edit</button>
            </div>
          </div>}
        </AccordionItemBody>
      </AccordionItem>
    );
  }
}


MenuAccordion.propTypes = {
  menu: PropTypes.object.isRequired,
  toggleUpdateModal: PropTypes.func.isRequired,
  toggleShowDeleteModal: PropTypes.func.isRequired,
  showOpsButtons: PropTypes.bool.isRequired,
  getMealsInMenu: PropTypes.func.isRequired,
};

export default MenuAccordion;
