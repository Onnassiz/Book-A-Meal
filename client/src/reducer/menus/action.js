import axios from 'axios';
import { setLoading, unsetLoading } from '../formState/action';

export const SET_MENUS = 'SET_MENUS';
export const SET_USER_MENUS = 'SET_USER_MENUS';
export const SET_MENUS_SERVER_ERRORS = 'SET_MENUS_SERVER_ERRORS';
export const SET_MENUS_ALERT = 'SET_MENUS_ALERT';

function setAlert(alert) {
  return dispatch => dispatch({
    type: SET_MENUS_ALERT,
    alert,
  });
}

export function setMenus(storeMenus, menu) {
  const menus = [menu].concat(storeMenus);
  return dispatch => dispatch({
    type: SET_MENUS,
    menus,
  });
}

function setMenusErrors(errors) {
  return dispatch => dispatch({
    type: SET_MENUS_SERVER_ERRORS,
    errors,
  });
}

function setMenusArray(storeMenus, menu) {
  const menus = storeMenus.concat(menu);
  return dispatch => dispatch({
    type: SET_MENUS,
    menus,
  });
}

function updateMenuAfterDelete(storeMenus, id) {
  const menus = storeMenus.filter(x => x.id !== id);
  return dispatch => dispatch({
    type: SET_MENUS,
    menus,
  });
}

export function updateMenuState(storeMenus, menu) {
  const index = storeMenus.findIndex(x => x.id === menu.id);
  const menus = storeMenus;
  menus[index] = menu;
  return dispatch => dispatch({
    type: SET_MENUS,
    menus,
  });
}

export function setClientMenus(menus, date) {
  return dispatch => dispatch({
    type: SET_USER_MENUS,
    menus,
    date,
  });
}

function addMealsFromMenuToArray(menus) {
  const meals = [];
  menus.forEach((item) => {
    item.meals.forEach((meal) => {
      meals.push({
        id: meal.id,
        name: meal.name,
        menuName: item.name,
        category: meal.category,
        description: meal.description,
        caterer: item.caterer,
        profileId: item.profileId,
        price: meal.price,
        totalPrice: meal.price,
        units: 1,
        menuId: item.id,
        imageUrl: meal.imageUrl,
      });
    });
  });
  return meals;
}

export function getUserMenus() {
  return (dispatch, getState) => {
    dispatch(setLoading());
    return axios.get('api/v1/menus/user').then((response) => {
      dispatch(unsetLoading());
      dispatch(setMenusArray(getState().menus.menus, response.data));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      // console.log(error.response.data);
      return error;
    });
  };
}


export function postMenu(menu) {
  return (dispatch, getState) => {
    dispatch(setLoading());
    return axios.post('api/v1/menus', menu).then((response) => {
      dispatch(unsetLoading());
      dispatch(setMenus(getState().menus.menus, response.data.menu));
      dispatch(setAlert('Menu successfully created'));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setMenusErrors(error.response));
      return error;
    });
  };
}

export function deleteMenuById(id) {
  return (dispatch, getState) => {
    dispatch(setLoading());
    return axios.delete(`api/v1/menus/${id}`).then((response) => {
      dispatch(unsetLoading());
      dispatch(updateMenuAfterDelete(getState().menus.menus, id));
      dispatch(setAlert('Meal successfully deleted'));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setMenusErrors(error.response.data));
      return error;
    });
  };
}

export function updateMenu(meal) {
  return (dispatch, getState) => {
    dispatch(setLoading());
    return axios.put(`api/v1/menus/${meal.id}`, meal).then((response) => {
      dispatch(unsetLoading());
      dispatch(updateMenuState(getState().menus.menus, response.data.menu));
      dispatch(setAlert('Menu successfully updated'));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setMenusErrors(error.response.data));
      return error;
    });
  };
}

export function getMenusByUnixTime(date) {
  let unixTime = new Date(date);
  unixTime = unixTime.setDate(unixTime.getDate()) / 1000;
  return (dispatch) => {
    dispatch(setLoading());
    return axios.get(`api/v1/menus/unixTime/${unixTime}`).then((response) => {
      dispatch(unsetLoading());
      dispatch(setClientMenus(addMealsFromMenuToArray(response.data), date));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setClientMenus([], date));
      return error;
    });
  };
}
