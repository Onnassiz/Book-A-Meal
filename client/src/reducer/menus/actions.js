import axios from 'axios';
import { setLoading, unsetLoading } from '../formState/actions';

export const SET_MENUS = 'SET_MENUS';
export const SET_USER_MENUS = 'SET_USER_MENUS';
export const SET_MENUS_SERVER_ERRORS = 'SET_MENUS_SERVER_ERRORS';
export const SET_MENUS_COUNT = 'SET_MENUS_COUNT';
export const SET_MENUS_AND_COUNT = 'SET_MENUS_AND_COUNT';

export function setMenus(storeMenus, newMenus) {
  const menus = newMenus.concat(storeMenus);
  return dispatch => dispatch({
    type: SET_MENUS,
    menus,
  });
}

export function setMenusCount(count) {
  return dispatch => dispatch({
    type: SET_MENUS_COUNT,
    count,
  });
}

function setMenusErrors(errors) {
  return dispatch => dispatch({
    type: SET_MENUS_SERVER_ERRORS,
    errors,
  });
}

// function setMenusArray(storeMenus, menus, pageNumber) {
//   const menusArray = storeMenus.menus.concat(menus.menus);
//   const { fetchedPages } = storeMenus;
//   fetchedPages.push({
//     page: pageNumber,
//     startId: menus.menus[0].id,
//   });
//   return dispatch => dispatch({
//     type: SET_MENUS_AND_COUNT,
//     menus: menusArray.sort(sortMenu),
//     count: menus.count,
//     fetchedPages,
//   });
// }

// function updateMenuAfterDelete(storeMenus, id) {
//   const menus = storeMenus.filter(x => x.id !== id);
//   return dispatch => dispatch({
//     type: SET_MENUS,
//     menus,
//   });
// }

export function updateMenuState(storeMenus, menu) {
  const index = storeMenus.findIndex(x => x.id === menu.id);
  const menus = storeMenus;
  menus[index] = menu;
  return dispatch => dispatch({
    type: SET_MENUS,
    menus,
  });
}

export function setClientMenus(menus, date) {
  return dispatch => dispatch({
    type: SET_USER_MENUS,
    menus,
    date,
  });
}

export function getUserMenus(offset = 0) {
  return (dispatch) => {
    dispatch(setLoading());
    return axios.get(`api/v1/menus/user/?offset=${offset}`).then((response) => {
      dispatch(unsetLoading());
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      return error;
    });
  };
}

export function getMealsInMenu(data, offset = 0, getAll = false) {
  const menu = data;
  const url = getAll ? `${menu.meals}?offset=${offset}` : `${menu.meals}?offset=${offset}&limit=5`;
  return (dispatch) => {
    dispatch(setLoading());
    return axios.get(url).then((response) => {
      dispatch(unsetLoading());
      menu.mealsArray = menu.mealsArray.concat(response.data);
      // dispatch(updateMenuState(getState().menus.menus, menu));
      return { response, menu };
    }).catch((error) => {
      dispatch(unsetLoading());
      return error;
    });
  };
}


export function postMenu(menu) {
  return (dispatch) => {
    dispatch(setLoading());
    return axios.post('api/v1/menus', menu).then((response) => {
      dispatch(unsetLoading());
      // dispatch(setMenus(getState().menus.menus, response.data.menus));
      // dispatch(setMenusCount(getState().menus.count + response.data.menus.length));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setMenusErrors(error.response.data));
      return error;
    });
  };
}

export function deleteMenuById(id) {
  return (dispatch) => {
    dispatch(setLoading());
    return axios.delete(`api/v1/menus/${id}`).then((response) => {
      dispatch(unsetLoading());
      // dispatch(updateMenuAfterDelete(getState().menus.menus, id));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setMenusErrors(error.response.data));
      return error;
    });
  };
}

export function updateMenu(menu) {
  return (dispatch) => {
    dispatch(setLoading());
    return axios.put(`api/v1/menus/${menu.id}`, menu)
      .then((response) => {
        // dispatch(updateMenuState(getState().menus.menus, response.data.menu));
        dispatch(unsetLoading());
        return response;
      }).catch((error) => {
        dispatch(setMenusErrors(error.response.data));
        dispatch(unsetLoading());
        return error;
      });
  };
}

export function getMealsInDailyMenu(date, offset = 0, limit = 12) {
  return (dispatch) => {
    dispatch(setLoading());
    return axios.get(`api/v1/meals/menus/?date=${date}&offset=${offset}&limit=${limit}`).then((response) => {
      dispatch(unsetLoading());
      // dispatch(setClientMenus(addMealsFromMenuToArray(response.data), date));
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setClientMenus([], date));
      return error;
    });
  };
}
