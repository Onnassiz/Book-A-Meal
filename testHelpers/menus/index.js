import { menu, meal } from '../../api/models';
import { insertMealMock } from '../meals/index';
import { getCatererId } from '../main';

const deleteMenus = (done, callDone = true) => menu.destroy({
  where: {},
}).then(() => meal.destroy({
  where: {},
}).then((deleted) => {
  if (callDone) {
    done();
  }
  return deleted;
}));

const getMealIds = () => insertMealMock().then((data) => {
  const meals = data.map(item => ({
    mealId: item.id,
    price: 1000,
  }));

  return meals.splice(20, 20);
});

const insertOneMenu = () => getCatererId().then((id) => {
  const newMenu = menu.build({
    userId: id,
    name: 'Fire bons',
    date: '2018-08-08',
  });

  return newMenu.save().then(data => data);
});

export {
  deleteMenus,
  getMealIds,
  insertOneMenu,
};
