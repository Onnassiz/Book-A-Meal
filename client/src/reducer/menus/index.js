import { SET_MENUS, SET_MENUS_ALERT, SET_MENUS_SERVER_ERRORS, SET_USER_MENUS } from './action';
import { getCurrentDate } from '../../utilities/functions';

const initialState = {
	menus: [],
	userMenus: [],
	currentDate: getCurrentDate(),
};

export default (state = initialState, action) => {
	switch (action.type) {
	case SET_MENUS:
		return { ...state, menus: action.menus };
	case SET_USER_MENUS:
		return { ...state, userMenus: action.menus, currentDate: action.date };
	case SET_MENUS_SERVER_ERRORS:
		return { ...state, errors: action.errors.data };
	case SET_MENUS_ALERT:
		return { ...state, alert: action.alert };
	default:
		return state;
	}
};
