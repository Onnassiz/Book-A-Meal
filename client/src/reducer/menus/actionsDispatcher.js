import axios from 'axios';
import { setLoading, unsetLoading } from '../formState/actionsDispatcher';
import { setMenusErrorsAction } from './actions';

const setMenusErrors = errors => dispatch => dispatch(setMenusErrorsAction(errors));

export const getUserMenus = (offset = 0) => (dispatch) => {
  dispatch(setLoading());
  return axios.get(`api/v1/menus/user/?offset=${offset}`).then((response) => {
    dispatch(unsetLoading());
    return response;
  }).catch((error) => {
    dispatch(unsetLoading());
    dispatch(setMenusErrors(error.response.data));
    return error;
  });
};

export const getMealsInMenu = (data, offset = 0, getAll = false) => {
  const menu = data;
  const url = getAll ? `${menu.meals}?offset=${offset}` : `${menu.meals}?offset=${offset}&limit=5`;
  return (dispatch) => {
    dispatch(setLoading());
    return axios.get(url).then((response) => {
      dispatch(unsetLoading());
      menu.mealsArray = menu.mealsArray.concat(response.data);
      return { response, menu };
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setMenusErrors(error.response.data));
      return error;
    });
  };
};


export const postMenu = menu => (dispatch) => {
  dispatch(setLoading());
  return axios.post('api/v1/menus', menu).then((response) => {
    dispatch(unsetLoading());
    return response;
  }).catch((error) => {
    dispatch(unsetLoading());
    dispatch(setMenusErrors(error.response.data));
    return error;
  });
};

export const deleteMenuById = id => (dispatch) => {
  dispatch(setLoading());
  return axios.delete(`api/v1/menus/${id}`).then((response) => {
    dispatch(unsetLoading());
    return response;
  }).catch((error) => {
    dispatch(unsetLoading());
    dispatch(setMenusErrors(error.response.data));
    return error;
  });
};

export const updateMenu = menu => (dispatch) => {
  dispatch(setLoading());
  return axios.put(`api/v1/menus/${menu.id}`, menu)
    .then((response) => {
      dispatch(unsetLoading());
      return response;
    }).catch((error) => {
      dispatch(unsetLoading());
      dispatch(setMenusErrors(error.response.data));
      return error;
    });
};

export const getMealsInDailyMenu = (date, offset = 0, limit = 12) => (dispatch) => {
  dispatch(setLoading());
  return axios.get(`api/v1/meals/menus/?date=${date}&offset=${offset}&limit=${limit}`).then((response) => {
    dispatch(unsetLoading());
    return response;
  }).catch((error) => {
    dispatch(unsetLoading());
    dispatch(setMenusErrors(error.response.data));
    return error;
  });
};
