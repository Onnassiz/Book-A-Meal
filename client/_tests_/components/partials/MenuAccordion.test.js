import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MenuAccordion from '../../../src/components/presentation/partials/MenuAccordion';
import { singleMenu } from '../../objectProps/menus.props';
import { getMealsInMenu } from '../../promiseProps/menus.props';


const props = {
  menu: singleMenu,
  toggleUpdateModal: jest.fn(),
  toggleShowDeleteModal: jest.fn(),
  showOpsButtons: true,
  getMealsInMenu,
};

const wrapper = shallow(<MenuAccordion {...props} />);

describe('<MenuAccordion />', () => {
  it('Should render MenuAccordion  component successfully', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should call handlePageChange and change current page to new page', () => {
    const handlePageChangeSpy = jest.spyOn(wrapper.instance(), 'handlePageChange');

    wrapper.instance().handlePageChange(3).then(() => {
      expect(wrapper.state().activePage).toBe(3);
      expect(handlePageChangeSpy).toHaveBeenCalled();
    });
  });

  it('Should call showMeals and display meals when show meals button is clicked', () => {
    const showMealsSpy = jest.spyOn(wrapper.instance(), 'showMeals');

    wrapper.instance().showMeals().then(() => {
      expect(wrapper.state().activePage).toBe(1);
      expect(showMealsSpy).toHaveBeenCalled();
    });
  });
});
