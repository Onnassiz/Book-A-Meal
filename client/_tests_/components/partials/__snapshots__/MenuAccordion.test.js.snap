// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuAccordion /> Should render MenuAccordion  component successfully 1`] = `
<AccordionItemWrapper
  accordionStore={
    AccordionContainer {
      "_listeners": Array [],
      "state": Object {
        "accordion": true,
        "items": Array [],
        "onChange": [Function],
      },
    }
  }
  className="accordion__item"
  disabled={false}
  expanded={false}
  hideBodyClassName=""
>
  <AccordionItemTitleWrapper
    className="accordion__title"
    hideBodyClassName=""
  >
    <h3
      className="u-position-relative"
    >
      Tue Sep 11 2018
      <div
        className="accordion__arrow"
        role="presentation"
      />
    </h3>
    <div
      style={
        Object {
          "fontSize": 13,
          "marginTop": 5,
        }
      }
    >
      8
       meals
    </div>
  </AccordionItemTitleWrapper>
  <AccordionItemBodyWrapper
    className="accordion__body"
    hideBodyClassName="accordion__body--hidden"
  >
    <div
      className="show-meals"
    >
      <button
        className="button"
        id="showMeals"
        onClick={[Function]}
      >
        <i
          className="ion-ios-eye"
        />
         Show Meals
      </button>
    </div>
    <div>
      <div
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <button
          className="button-error"
          onClick={[MockFunction]}
          style={
            Object {
              "width": 110,
            }
          }
        >
          <i
            className="ion-ios-trash"
          />
           Delete
        </button>
        <button
          className="button"
          onClick={[MockFunction]}
          style={
            Object {
              "width": 110,
            }
          }
        >
          <i
            className="ion-ios-compose-outline"
          />
           Edit
        </button>
      </div>
    </div>
  </AccordionItemBodyWrapper>
</AccordionItemWrapper>
`;
